### PR TITLE
Org identity trigger update

### PR DESCRIPTION
**Pull request by xnetsrak**
## Description

Added timer trigger for updated Organization name from CVR lookup.
Email notification on change.

## References

* #0000

### CODE REVIEW

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Organization/UpdateOrganizationIdentityTriggerHandler.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    public async Task<Unit> Handle(UpdateOrganisationIdentityTriggerCommand request, CancellationToken cancellationToken)
+    {
+        var allOrganizations = await _organizationRepository
+            .GetAsync()
+            .ConfigureAwait(false);
+
+        var organizationsToCheck = allOrganizations
+            .Where(e =>
+                !string.IsNullOrWhiteSpace(e.BusinessRegisterIdentifier.Identifier) &&
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Organization/UpdateOrganizationIdentityTriggerHandler.cs:58]: CVR is mandatory, right?

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Organization/UpdateOrganizationIdentityTriggerHandler.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                    .StartsWith("ENDK", StringComparison.InvariantCultureIgnoreCase));
+
+        foreach (var organization in organizationsToCheck)
+        {
+            var identityResponse = await _organizationIdentityRepository
+                .GetAsync(organization.BusinessRegisterIdentifier)
+                .ConfigureAwait(false);
+
+            if (identityResponse == null) continue;
+            if (string.IsNullOrWhiteSpace(identityResponse.Name) || organization.Name == identityResponse.Name) continue;
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Organization/UpdateOrganizationIdentityTriggerHandler.cs:69]: No need for whitespace check, equality is enough. Also, should we have culturesensitive string comparison here?

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Organization/UpdateOrganizationIdentityTriggerHandler.cs (hunk 1 above)
Comment (xnetsrak) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Organization/UpdateOrganizationIdentityTriggerHandler.cs:58]: yes

Comment (defectiveAi) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Organization/UpdateOrganizationIdentityTriggerHandler.cs:58]: Oki, chop IsNullOrWhitespace.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Organization/UpdateOrganizationIdentityTriggerHandler.cs (hunk 2 above)
Comment (xnetsrak) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Organization/UpdateOrganizationIdentityTriggerHandler.cs]: better ?

**Review by defectiveAi — APPROVED** [review]

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**